src/components/PeoplePage.tsx

```diff
@@ -1,32 +1,109 @@
+import React, { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+
+export const PeoplePage: React.FC<{}> = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  let visiblePeople = [...people];
+
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || null;
+  const centuries = searchParams.getAll('centuries') || [];
+  const sorting = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  useEffect(() => {
+    const loadPeople = async () => {
+      try {
+        setPeople(await getPeople());
+      } catch (err) {
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadPeople();
+  }, []);
+
+  const isErrorMessageVisible = !isLoading && error;
+  const isPeopleTableVisible = !isLoading && !error && people.length > 0;
+
+  if (sex) {
+    visiblePeople = visiblePeople.filter(person => person.sex === sex);
+  }
+
+  if (query) {
+    visiblePeople = visiblePeople
+      .filter(person => {
+        const searchFilter = `${person.motherName?.toLowerCase()}${person.fatherName?.toLowerCase()}${person.name.toLowerCase()}`;
+
+        return searchFilter.includes(query.toLowerCase());
+      });
+  }
+
+  if (centuries.length > 0) {
+    visiblePeople = visiblePeople.filter(person => {
+      return centuries
+        .includes(String(Math.floor((person.born - 1) / 100) + 1));
+    });
+  }
+
+  if (sorting) {
+    switch (sorting) {
+      case 'name':
+      case 'sex':
+        visiblePeople = visiblePeople.sort((a, b) => {
+          return a[sorting].localeCompare(b[sorting]);
+        });
+        break;
+      case 'born':
+      case 'died':
```

Comment:
It would be cool not to use magic values and put such strings into constants and reuse them:
```
const NAME = 'name';
const SEX = 'sex';
const BORN= 'born';
const DIED= 'died';
```
This will increase the readability of the code and save us from possible errors.